src/acl.c

```diff
@@ -2910,9 +2910,18 @@ void aclCommand(client *c) {
             }
         }
         raxStop(&ri);
-    } else if (!strcasecmp(sub, "whoami") && c->argc == 2) {
+    } else if (!strcasecmp(sub, "whoami") && (c->argc == 2 || c->argc == 3)) {
         if (c->user != NULL) {
-            addReplyBulkCBuffer(c, c->user->name, sdslen(c->user->name));
+            if (c->argc == 3) {
```

Comment:
I generally don't like variadic arguments that change the type of response. So would also somewhat prefer another command.

Also, do we want this to be a variant of `ACL GETUSER` instead? Somewhat related to this issue, https://github.com/valkey-io/valkey/issues/1807.

Reply:
Another command? you mean a new command?  I am afraid the user need to remember one more. 
I guess this command is not same as ACL GETUSER, because this command only apply to current connection, even user do not know the username either.

Reply:
> I guess this command is not same as ACL GETUSER, because this command only apply to current connection, even user do not know the username either.

A new command, like `ACL GETMYSELF`, is what I'm suggesting. I agree it's one more command.